api/server/httprest/server.go

```diff
@@ -44,11 +45,10 @@ func New(ctx context.Context, opts ...Option) (*Server, error) {
 	}
 	var handler http.Handler
 	defaultReadHeaderTimeout := time.Second
+	handler = middleware.MiddlewareChain(g.cfg.router, g.cfg.middlewares)
 	if g.cfg.timeout > 0*time.Second {
 		defaultReadHeaderTimeout = g.cfg.timeout
 		handler = http.TimeoutHandler(g.cfg.router, g.cfg.timeout, "request timed out")
```

Comment:
we need to replace g.cfg.router with handler in this case

Reply:
But we are wrapping the mux with middlewares why do we wanna wrap http.Handler instead